### PR TITLE
GUVNOR-3168 GUVNOR-3167 [DMN Editor] Write unmarshaller marshaller

### PR DESCRIPTION
**Pull request by tarilabs**
Early to be merged, just sharing to validate the current ongoing approach... @manstis can you kindly try it and give me some feedback, please?

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:

- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/476
- https://github.com/kiegroup/kie-wb-common/pull/986

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java [hunk 1]
```diff
@@ -17,37 +17,139 @@
...
+                    .asNode();
+            ((View) node.getContent()).setDefinition( org.kie.workbench.common.dmn.api.definition.v1_1.Decision.from( (Decision) dmn ) );
+            return node;
+        } else {
+            throw new UnsupportedOperationException("TODO"); // TODO 
+        }
+    }
+    
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private void connectRootWithChild(Node dmnDiagramRoot, Node child) {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java]: The same method declared twice in the PR, can be moved to some Util class as a static method.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java [hunk 2]
```diff
@@ -17,37 +17,139 @@
...
+        final String uuid = org.kie.workbench.common.stunner.core.util.UUID.uuid();
+        final Edge<Child, Node> edge = new EdgeImpl<>(uuid);
+        edge.setContent(new Child());
+        connectEdge(edge, dmnDiagramRoot, child);
+    }
+    
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private void connectEdge( Edge edge, Node source, Node target ) {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java]: The same method declared twice in the PR, can be moved to some Util class as a static method.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNUnmarshallTest.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            result.setGraph(graph);
+            return result;
+        }).when(applicationFactoryManager).newDiagram(anyString(),
+                                                      anyString(),
+                                                      any(Metadata.class));
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Test
+    public void test2() throws IOException {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNUnmarshallTest.java]: Please add some assertion next to `println()`

**Comment by jomarko** [comment]
@tarilabs sorry, I haven't notice it is just for showing the progress. I will return once it is done. Anyway, thanks for implementing.

**Comment by tarilabs** [comment]
To avoid misunderstanding I've renamed the PR title.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNUnmarshallTest.java [hunk 4]
```diff
@@ -142,7 +140,14 @@ public void setup() throws Exception {
             String uuid = (String) invocationOnMock.getArguments()[0];
             String id = (String) invocationOnMock.getArguments()[1];
             if (DMNDefinitionSet.class.getName().equals(id)) {
-                Graph graph = (Graph) dmnGraphFactory.build(uuid, DMN_DEF_SET_ID);
+                // TODO this is different from the stunner jbpm test which this dmn test is based on
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNUnmarshallTest.java]: Yes, I wondered too... however looking at the tests it appears the BPMN2 [definition](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/resources/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/basic.bpmn#L32) contains a "diagram" element whereas DMN does not.

Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNUnmarshallTest.java]: just added as comment in order not to forget.
Anyway bigger problem I recompiled everything but still I get "Uncaught error on client side: (TypeError) : Cannot read property 'asString_1_g$' of null" on this side :(

Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNUnmarshallTest.java]: Yes, I wondered too. However the BPMN2 unmarshalling is different to DMN's!

It appears that the BPMN2 unmarshaller creates the "diagram" element and adds it to the graph [elsewhere](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/marshall/json/builder/BPMNGraphGenerator.java#L165).

**Comment by tarilabs** [comment]
```
12:37:38 [ERROR] /home/jenkins/workspace/kie-wb-common-pullrequests/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-api/src/main/java/org/kie/workbench/common/stunner/cm/definition/ReusableSubprocess.java:[49,1] Internal error in org.kie.workbench.common.forms.adf.processors.FormDefinitionsProcessorjava.lang.NoClassDefFoundError: Could not initialize class freemarker.template.Configuration
12:37:38 [ERROR] at org.kie.workbench.common.forms.adf.processors.FormDefinitionsProcessor.writeTemplate(FormDefinitionsProcessor.java:674)
12:37:38 [ERROR] at org.kie.workbench.common.forms.adf.processors.FormDefinitionsProcessor.extracFormFields(FormDefinitionsProcessor.java:633)
...
12:37:38 [ERROR] at java.util.concurrent.FutureTask.run(FutureTask.java:266)
12:37:38 [ERROR] at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
12:37:38 [ERROR] at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
12:37:38 [ERROR] at java.lang.Thread.run(Thread.java:748)
12:37:38 [ERROR] Caused by: java.lang.RuntimeException: Failed to load and parse freemarker/version.properties
12:37:38 [ERROR] at freemarker.template.Configuration.<clinit>(Configuration.java:461)
12:37:38 [ERROR] ... 38 more
12:37:38 [ERROR] Caused by: java.util.zip.ZipException: ZipFile closed
12:37:38 [ERROR] at java.util.zip.ZipFile.ensureOpenOrZipException(ZipFile.java:679)
12:37:38 [ERROR] at java.util.zip.ZipFile.access$1300(ZipFile.java:60)
12:37:38 [ERROR] at java.util.zip.ZipFile$ZipFileInputStream.read(ZipFile.java:716)
12:37:38 [ERROR] at java.util.zip.ZipFile$ZipFileInflaterInputStream.fill(ZipFile.java:419)
12:37:38 [ERROR] at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:158)
12:37:38 [ERROR] at java.io.FilterInputStream.read(FilterInputStream.java:133)
12:37:38 [ERROR] at java.io.FilterInputStream.read(FilterInputStream.java:107)
12:37:38 [ERROR] at java.util.Properties$LineReader.readLine(Properties.java:435)
12:37:38 [ERROR] at java.util.Properties.load0(Properties.java:353)
12:37:38 [ERROR] at java.util.Properties.load(Properties.java:341)
12:37:38 [ERROR] at freemarker.template.Configuration.<clinit>(Configuration.java:436)
12:37:38 [ERROR] ... 38 more
12:37:38 [ERROR] -> [Help 1]
```

**Comment by manstis** [comment]
Jenkins please retest this (builds and runs OK locally)

**Comment by tarilabs** [comment]
c82de4d unrelated failure:
```
12:10:27 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:testCompile (default-testCompile) on project guvnor-ala-build-maven: Compilation failure: Compilation failure:
12:10:27 [ERROR] /home/jenkins/workspace/kie-wb-common-pullrequests/upstream-repos/guvnor/guvnor-ala/guvnor-ala-build-maven/src/test/java/org/guvnor/ala/build/maven/executor/RepositoryVisitorTest.java:[46,49] cannot find symbol
12:10:27 [ERROR] symbol:   class JGitUtil
12:10:27 [ERROR] location: package org.uberfire.java.nio.fs.jgit.util
12:10:27 [ERROR] /home/jenkins/workspace/kie-wb-common-pullrequests/upstream-repos/guvnor/guvnor-ala/guvnor-ala-build-maven/src/test/java/org/guvnor/ala/build/maven/executor/RepositoryVisitorTest.java:[46,1] static import only from classes and interfaces
12:10:27 [ERROR] -> [Help 1]
12:10:27 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:testCompile (default-testCompile) on project guvnor-ala-build-maven: Compilation failure
```

**Comment by tarilabs** [comment]
Requiring DROOLS-1682 to be merged first.

**Comment by jomarko** [comment]
Hi @tarilabs , please what is status of this PR, can I have look now?

**Comment by tarilabs** [comment]
@jomarko as far as I know that is not a strict enforcing rule for our code style. So I (as others, I think) apply that when needed to avoid code becoming unreadable - although I'm aware legibility " is in the eye of the beholder"

**Comment by tarilabs** [comment]
Requires first to merge DROOLS-1682

Provides:
- node marshal/unmarshal for DRG nodes as: Input Data, Decision, Business Knowledge Model, Knowledge Source
. edge marshal/unmarshal for DRG relationship as: Information Requirement, Knowledge Requirement, Authority Requirement
- Artifact node and Artifact edge marshal/unmarshal for additional DRG Artifact element as Text Annotation (node), Association (edge)
- Support for basic LiteralExpression which, although not editable yet from the stunner ui, it can be helpful when "importing" an existing .dmn and marshal it back to .dmn
- Perform the Stunner Graph node and edge test assertions as indicated by @manstis 
- Include all fixes provided through all the feedback so far on this task by @manstis 
- Perform DMN runtime execution test for `diamond.dmn` provided model after a marshal/unmarshal round-trip in stunner, to ensure consistency

/cc @etirelli

**Comment by jomarko** [comment]
@tarilabs I think codestyles in [1] were introduced to be used in kiegroup projects to eliminate formatting changes in PRs. It is natural that every developer has slightly different needs to format code. Someone like spaces between brackets, someone fluent api calls on one line, someone fluent api calls on multiple lines etc. However these [1] codestyles are some compromise. The consistency in code style that they brings, helps to read PRs. 

I am not sure if they are strict, maybe @hasys would know as one of iniciators, I just wanted to say that I am using this code style because of the above, and I see it also in PRs by other developers, that is why I expected it also here.

Thanks a lot for this PR. I will have a look.

[1]
https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/master/ide-configuration

**Comment by tarilabs** [comment]
@jomarko my perspective is that a little flexibility should go the possible ways; by the same principle you enlist above if I check the same [1] then multiple parameters should NOT go on multiple lines. But IntelliJ users tend to prefer sometimes something like:
```
new MyClass(f,
            x,
            null,
            null,
            3,
            null,
            a);
```
Simply because the IntelliJ debugger leverage this layout, and also the IntelliJ editor show the parameters name in overlay.
There are not equivalent Eclipse features, is NOT the common code style, but personally I try to adapt.
Although there is a substantial difference, in this PR case space before first parameter is not that relevant, while the counter example I provided above is considered bad practice, for instance referring to "Effective Java" book by Joshua Bloch because a fluent builder would have been more appropriate.
In other words: I see your point about space before first and last parameters; but I propose we should focus on more priority issues.

**Comment by hasys** [comment]
Hi @tarilabs,
I am agree with you:

> I propose we should focus on more priority issues.

Exactly for this aim in Stunner, new Forms, current Designer we just import IDEA style from [droolsjbpm-build-bootstrap](https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/master/ide-configuration/intellij-configuration/code-style) and using it as is EVERYWHERE. Why? Because:

> I (as others, I think) apply that when needed to avoid code becoming unreadable - although I'm aware legibility " is in the eye of the beholder"

Is too uncertain and we had a lot of useless style changes before in, almost, every singl PR. We don't want to see it again (style changes hard to review, lead to comments like: "Please, rebase it, you missed/added redundant space", it is easy to miss important change in PR with a lot of code style changes and so on).

Yes, current style is not sexy for 100% situations and 100% developers, but

> but I propose we should focus on more priority issues.

Is more important than discussions about were addition spaces improve readability for a bit.

Thank you, @tarilabs.

**Comment by tarilabs** [comment]
@hasys thank you for the feedback but this is avoiding my main point:
that I allow for flexibility when code incoming from IntelliJ does not comply strictly with code style but I let it pass anyway,
while none flexibility is given in return now?

**Comment by hasys** [comment]
@tarilabs,
we had code style flexibility and everyone had different points about what is more readable code. As a bonus we had **not readable** PRs.

You can have your own local style for IntelliJ IDEA with any level of flexibility, just convert all code to your own the best one code style before coding (it is about tree mouse click in IntelliJ IDEA), but it is extremely useful for everyone when code style in repository is unified.

So please, fill free to use "Reformat Code" feature before development (to your own the best one Code Style) and after development (to KIE Group default one) to help others review PRs and avoid Code Styles wars in the future.

P.S. KIE Group default style for IntelliJ IDEA is import [droolsjbpm-build-bootstrap](https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/master/ide-configuration/intellij-configuration/code-style) and don't do nothing with it.

**Comment by tarilabs** [comment]
@hasys possibly I was not able to fully express my perspective.

I'm not arguing at all which is the best code style.
I am aware droolsjbpm-build-bootstrap formalizes a code style, and I understand is the common agreed compromise.
I acknowledge that regardless which parts I like or dislike. Because I also believe agreement for collaboration trumps personal preferences for code style.
I have imported droolsjbpm-build-bootstrap as per readme and configured it inside my Eclipse. Where the tooling does not do automatically, I try to comply manually as best as I can.
Naturally, some times it might missed on a few lines.

My point is that I don't enforce _strictly_ when code which comes from other IDE users, like IntelliJ, do NOT comply with code style, when I can see it provide some benefit to the original commiter; I just try to adapt myself to the different situations, up to a reasonable degree. I find a bit disappointing when this is not mutually returned, like this case? Especially considering the bigger picture. The conclusion I would take from this discussion is that I'm wrong in allowing flexibility to pass through.

Anyway, I just wanted to be crystal clear to express my PoV; I'm sorry if this is seen as a potential for "code style wars" this is not really my intention.

Thank you for your patience in listening to my full explanation. I hope it could be considered for future discussion on this topic.

I will comply with strict enforcement requested, I will submit a subsequent commit in this PR which enforces the droolsjbpm-build-bootstrap code style.

**Comment by tarilabs** [comment]
@hasys @jomarko done commit 494ef28 strictly enforces droolsjbpm-build-bootstrap common code style on the classes created or heavily modified subject of this PR.

FYI in packages:
```
org.kie.workbench.common.dmn.api.definition.v1_1
org.kie.workbench.common.dmn.api.factory
```
just to name a few previously existing classes like:
```
BusinessKnowledgeModel,
DMNDiagram,
DMNGraphFactoryImpl
```
just to name a few do NOT comply strictly with code style.

In other words: they apply IntelliJ convenience for parameters which is not compliant with the droolsjbpm-build-bootstrap common code style.

Although the flexibility I practice while reviewing code I understand is not mutually returned, I'm not going to strictly enforce code style on previously existing code which I just happen to modify in part.

**Comment by hasys** [comment]
@tarilabs, thank you for detailed answer. I see your pain point. Unfortunately it is a real pain if you are using Eclipse right now. I will unify Eclipse and IDEA styles soon. (I didn't it before due to no one of Designers/new Forms developers using Eclipse). Sorry for inconvenience.

**Comment by tarilabs** [comment]
@hasys no problems I don't think is a urgent thing because my contribution (as one of the possibly few Eclipse user) to kie-wb-common is somewhat limited; meanwhile code style as far as I can see is not enforced _strictly_ in the others kiegroup repos and to be honest I'm much more happy that way :)

**Comment by manstis** [comment]
@jomarko Let's review this at a functional level. I will submit another PR (once this is merged) with "InteliJ code-formatting" and everyone can be happy ;-)

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java [hunk 5]
```diff
@@ -17,47 +17,323 @@
...
+        this.textAnnotationConverter = new TextAnnotationConverter(factoryManager);
+        this.marshaller = DMNMarshallerFactory.newDefaultMarshaller();
     }
 
-    @Override
-    @SuppressWarnings("unchecked")
-    public Graph unmarshall(final Metadata metadata,
-                            final InputStream input) throws IOException {
+    @Deprecated
+    public Graph unmarshallFromStunnerJSON(final Metadata metadata, final InputStream input) throws IOException {
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java]: We agreed with @manstis to keep it as an internal utility method meanwhile all the UI artifacts are implemented, so to help with the marshaller work.

File: kie-wb-common-dmn/kie-wb-common-dmn-api/src/main/java/org/kie/workbench/common/dmn/api/definition/v1_1/BusinessKnowledgeModel.java [hunk 6]
```diff
@@ -63,8 +63,9 @@
     @FormField(afterElement = "name")
     protected InformationItem variable;
 
-    @PropertySet
-    @FormField(afterElement = "variable")
+    // TODO this was disabled during marshaller work task.
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-api/src/main/java/org/kie/workbench/common/dmn/api/definition/v1_1/BusinessKnowledgeModel.java]: ah sorry I misunderstood. Removed in upcoming commit.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java [hunk 7]
```diff
@@ -17,47 +17,323 @@
...
+                        Node requiredNode = elems.get(reqInputID).getValue();
+                        Edge myEdge = factoryManager.newElement(UUID.uuid(), org.kie.workbench.common.dmn.api.definition.v1_1.AuthorityRequirement.class).asEdge();
+                        connectEdge(myEdge, requiredNode, currentNode);
+                        setConnectionMagnets(myEdge);
+                    }
+                }
+            }
+        }
+
+        Map<String, Node<View<TextAnnotation>, ?>> textAnnotations = dmnXml.getArtifact().stream().filter(org.kie.dmn.model.v1_1.TextAnnotation.class::isInstance).map(org.kie.dmn.model.v1_1.TextAnnotation.class::cast).collect(Collectors.toMap(org.kie.dmn.model.v1_1.TextAnnotation::getId, textAnnotationConverter::nodeFromDMN));
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java]: Nearly most Java developer advocates for having it on multiple line, most Java books and manuals have it on multiple lines, but QE peers insisted for _strict_ `ctrl+shift+F` "EVERYWHERE"(cit.)...  so I win that prize only by accident :P

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java [hunk 8]
```diff
@@ -17,47 +17,323 @@
...
+            if (node.getContent() instanceof View<?>) {
+                View<?> view = (View<?>) node.getContent();
+                if (view.getDefinition() instanceof DRGElement) {
+                    DRGElement n = (org.kie.workbench.common.dmn.api.definition.v1_1.DRGElement) view.getDefinition();
+                    nodes.put(n.getId().getValue(), stunnerToDMN(node));
+                } else if (view.getDefinition() instanceof TextAnnotation) {
+                    TextAnnotation textAnnotation = (TextAnnotation) view.getDefinition();
+                    textAnnotations.put(textAnnotation.getId().getValue(), textAnnotationConverter.dmnFromNode((Node<View<TextAnnotation>, ?>) node));
+
+                    // TODO no way to query graph for Edges?
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshaller.java]: As per our email discussion we clarified there is no API to query graph for edges, beside knowing IDs, hence I'm going to remove that comment in subsequent commit.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/AssociationConverter.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            Node<?, ?> targetNode = e.getTargetNode();
+            if (targetNode.getContent() instanceof View<?>) {
+                View<?> view = (View<?>) targetNode.getContent();
+                if (view.getDefinition() instanceof DRGElement) {
+                    DRGElement drgElement = (DRGElement) view.getDefinition();
+                    org.kie.dmn.model.v1_1.DMNElementReference targetRef = new org.kie.dmn.model.v1_1.DMNElementReference();
+                    targetRef.setHref(new StringBuilder("#").append(drgElement.getId().getValue()).toString());
+
+                    org.kie.dmn.model.v1_1.Association adding = new org.kie.dmn.model.v1_1.Association();
+                    // TODO missing id, name
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/AssociationConverter.java]: sorry spurious comment, good catch, in this case was needed only ID indeed.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/LiteralExpressionPropertyConverter.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+public class LiteralExpressionPropertyConverter {
+
+    public static LiteralExpression wbFromDMN(org.kie.dmn.model.v1_1.LiteralExpression dmn) {
+        Id id = new Id(dmn.getId());
+        Description description = new Description(dmn.getDescription());
+        QName typeRef = QNamePropertyConverter.wbFromDMN(dmn.getTypeRef());
+        Text text = new Text(dmn.getText());
+        ExpressionLanguage expressionLanguage = new ExpressionLanguage(dmn.getExpressionLanguage());
+        // TODO missing importedValues
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/LiteralExpressionPropertyConverter.java]: We are currently not supporting semantically that from DMN side, if I recall correctly. Anyway I've added the converter support, coming in subsequent commit.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/LiteralExpressionPropertyConverter.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public static org.kie.dmn.model.v1_1.LiteralExpression dmnFromWB(LiteralExpression wb) {
+        org.kie.dmn.model.v1_1.LiteralExpression result = new org.kie.dmn.model.v1_1.LiteralExpression();
+        result.setId(wb.getId().getValue());
+        result.setDescription(wb.getDescription().getValue());
+        QNamePropertyConverter.setDMNfromWB(wb.getTypeRef(), result::setTypeRef);
+        result.setText(wb.getText().getValue());
+        result.setExpressionLanguage(wb.getExpressionLanguage().getValue());
+        // TODO missing importedValues
```
Comment (tarilabs) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/LiteralExpressionPropertyConverter.java]: above

**Comment by tarilabs** [comment]
@manstis thank you for your feedback implemented with d2da864

> Use of extensions for UI aspects in DMN 1.1
https://issues.jboss.org/browse/GUVNOR-3352

> Other expression types
https://issues.jboss.org/browse/GUVNOR-3353

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by tarilabs** [comment]
@jomarko thanks fixed with 6924dc3

**Comment by manstis** [comment]
@jomarko Please make sure you're not seeing an occurrence of this: https://issues.jboss.org/browse/GUVNOR-3217 (i.e. the properties you _think_ you've updated are not actually the properties that _were_ updated).

**Comment by tarilabs** [comment]
Can't reproduce, for example Description:
image

I also retested locally for Name, it get persisted in dmn xml. Then I close the webapp. I open the webapp. I load diagrams from server button. I select diagram, I click Edit button. My modification are reloaded as I would expect.

**Comment by manstis** [comment]
Jenkins please execute full downstream build

**Comment by tarilabs** [comment]
the downstream error is:
```
13:50:00 [INFO] Kie Workbench - Common - Project Imports Editor Client FAILURE [  1.330 s]
...
13:50:01 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:compile (default-compile) on project kie-wb-common-project-imports-editor-client: Compilation failure: Compilation failure:
13:50:01 [ERROR] /home/jenkins/workspace/kie-wb-common-downstream-pullrequests/kie-wb-common-screens/kie-wb-common-project-imports-editor/kie-wb-common-project-imports-editor-client/src/main/java/org/kie/workbench/common/screens/projectimportsscreen/client/forms/ProjectImportsScreenPresenter.java:[47,1] Internal error in org.uberfire.annotations.processors.WorkbenchEditorProcessorjava.lang.NoClassDefFoundError: org.uberfire.annotations.processors.AbstractGenerator
13:50:01 [ERROR] at org.uberfire.annotations.processors.WorkbenchEditorProcessor.<init>(WorkbenchEditorProcessor.java:48)
```

**Comment by manstis** [comment]
Jenkins please execute full downstream build

**Comment by manstis** [comment]
@tarilabs Yep.. one of our long running, sporadically re-occurring build issues (perhaps one day, when nobody has anything else to do someone will look into it.. however atm whilst a _known_ problem; nobody - me included - wants to invest the time in fixing when Jenkins is our slave!)

**Comment by psiroky** [comment]
@manstis @tarilabs I've actually spent hours trying to fix this few months back. I was not able to find a definitive fix and so at some point I just gave up. I am not planning to look at this again as I am out of ideas.

**Comment by tarilabs** [comment]
:tada:
